packages/common/src/store/upload/reducer.ts

```diff
@@ -59,12 +59,22 @@ const initialUploadState: ProgressState = {
   },
   stems: []
 }
-const getInitialProgress = (upload: TrackForUpload | StemUploadWithFile) => {
+const getInitialProgress = (
+  upload: TrackForUpload | StemUploadWithFile | StemUploadWithFile
+) => {
   const res = cloneDeep(initialUploadState)
   res.art.total =
     'artwork' in upload.metadata ? upload.metadata.artwork?.file?.size ?? 0 : 0
   res.audio.total = upload.file?.size ?? 0
-  res.stems = upload.metadata.stems?.map(getInitialProgress) ?? []
+  res.stems =
```

Comment:
bit confused by this - it's a recursive call to get the progress of each stem as well as the parent track?
would this not work?
```
res.stems = upload.metadata?.stems
          .map((stem) => {
            return 'file' in stem ? getInitialProgress : null
          }
  ) ?? []
```
and then filter nullable? also not great but maybe easier to read and avoid ts-ignore?

Reply:
the type of upload.metadata doesn't necessarily have stems on it because the upload metadata could be of type `StemUploadWithFile` so we have to use the `'stems' in upload.metadata` check to narrow the type

found a slightly cleaner way with a cast instead of ts-ignore though